Update PR #1061 with the changes discussed in the FuSa call to make t…

nicpappler opened the pull request:
Update PR #1061 with the changes discussed in the FuSa call to make t…he profile more reality friendly and implementable by tool creators

This updates the PR #1061

nicpappler committed:
Update PR #1061 with the changes discussed in the FuSa call to make the profile more reality friendly and implementable by tool creators

nicpappler committed:
adding new relationships

nicpappler committed:
incorporating Steven's feedback during the FuSa call

stevenc-stb commented on model/Core/Properties/devLifeCycleStage.md:
```diff
@@ -0,0 +1,17 @@
+## Metadata
+
+- name: devLifeCycleStage
+- Nature: RequirementProperty
```
```suggestion
- Nature: DataProperty
```
- Nature:  are DataProperty or ObjectProperty  see [format](https://github.com/spdx/spdx-3-model/blob/develop/docs/format.md#properties)

stevenc-stb commented on model/Core/Properties/requirementRational.md:
```diff
@@ -0,0 +1,17 @@
+## Metadata
+
+- name: requirementRational
+- Nature: RequirementProperty
```
Nature: are DataProperty or ObjectProperty see [format](https://github.com/spdx/spdx-3-model/blob/develop/docs/format.md#properties)
```suggestion
- Nature: DataProperty
```

stevenc-stb commented on model/Core/Properties/requirementStatement.md:
```diff
@@ -0,0 +1,17 @@
+## Metadata
+
+- name: requirementStatement
+- Nature: RequirementProperty
```
```suggestion
- Nature: DataProperty
```

stevenc-stb commented on model/Core/Properties/requirementUUID.md:
```diff
@@ -0,0 +1,17 @@
+## Metadata
+
+- name: requirementUUID
+- Nature: RequirementProperty
```
```suggestion
- Nature: DataProperty
```

stevenc-stb reviewed (changes requested):
some Markdown fixes

Nature: is DataProperty or ObjectProperty see [format](https://github.com/spdx/spdx-3-model/blob/develop/docs/format.md#properties)

stanislaw commented on model/Core/Properties/requirementRational.md:
```diff
@@ -0,0 +1,17 @@
+SPDX-License-Identifier: Community-Spec-1.0
+
+# requirementRational
```
Should this be `requirementRationale`? If yes, then, also the file name change is needed.

stanislaw commented on model/Core/Classes/Requirement.md:
```diff
@@ -0,0 +1,36 @@
+
+## Description
+
+A requirement element is a distinct article or unit defining an expectation, need, behaviour, design intent etc. of an item that already exists or is to be created based on this requirement.
```
```suggestion
A requirement element is a distinct unit that defines an expectation, need, behavior, or design intent of an item that either already exists or is to be created in accordance with this requirement.
```

stanislaw replied:
```suggestion
A requirement element is a distinct article or unit defining an expectation, need, behaviour, design intent, etc., of an item that already exists or is to be created based on this requirement.
```

stanislaw commented on model/Core/Classes/Requirement.md:
```diff
@@ -0,0 +1,36 @@
+
+## Summary
+
+A distinct unit representing a requirement as it is used by systems, software and hardware engineering.
```
```suggestion
A distinct unit representing a requirement, as used in systems, software, and hardware engineering.
```

stanislaw commented on model/Core/Properties/requirementRational.md:
```diff
@@ -0,0 +1,17 @@
+
+## Description
+
+Additional detail used to define the reason/rational, why this requirement is there or give some additional information, needed to understand the requirement. Usually less formal than the requirement wording of the RequirementStatement.
```
Here I would add:

> Additional detail used to define the reason or justification for the existence of the requirement. The rationale is usually less formal than the wording of the requirement statement itself. 
>
> If a requirement is linked to a parent requirement, the rationale shall provide a high-level justification describing how the child requirement satisfies or contributes to fulfilling the intent of the parent requirement. If a requirement has no parent linkage, the rationale shall provide a self-contained explanation of the underlying reasoning or purpose for the requirement — i.e., it shall justify why the requirement is stated as it is.

Note that I removed the `additional information` because this can be done by using the parent SPDX element fields.

stanislaw commented on model/Core/Properties/requirementUUID.md:
```diff
@@ -0,0 +1,17 @@
+
+## Summary
+
+Provides an universally unique Requirement ID.
```
```suggestion
Provides a universally unique Requirement ID.
```

stanislaw commented on model/Software/Vocabularies/ContentIdentifierType.md:
```diff
@@ -18,3 +18,4 @@ ContentIdentifierType specifies the type of a content identifier.
 
 - gitoid: [Gitoid](https://www.iana.org/assignments/uri-schemes/prov/gitoid), stands for [Git Object ID](https://git-scm.com/book/en/v2/Git-Internals-Git-Objects). A gitoid of type blob is a unique hash of a binary artifact. A gitoid may represent either an [Artifact Identifier](https://github.com/omnibor/spec/blob/eb1ee5c961c16215eb8709b2975d193a2007a35d/spec/SPEC.md#artifact-identifier-types) f ...
 - swhid: SoftWare Hash IDentifier, a persistent intrinsic identifier for digital artifacts, such as files, trees (also known as directories or folders), commits, and other objects typically found in version control systems. The format of the identifiers is defined in the [SWHID specification](https://www.swhid.org/specification/v1.1/4.Syntax) (ISO/IEC DIS 18670). They typically look like `swh:1:c ...
+- reqUID: the UID used by a requirements management or any other lifecycle management tool to uniquely identify a requirement item. UID, or unique ID, is a standard term in requirements engineering.
```
Should this be `reqUID` or `reqUUID`? I think the addition of `universally unique` is very important here.

Also, should it be written as `requirementUUID`?

stanislaw commented on model/Core/Vocabularies/RelationshipType.md:
```diff
@@ -81,6 +81,8 @@ name completes the sentence:
 - republishedBy: Designates a `from` Vulnerability's details were tracked, aggregated, and/or enriched to improve context (i.e. NVD) by each `to` Agent.
 - serializedInArtifact: The `from` SpdxDocument can be found in a serialized form in each `to` Artifact.
 - testedOn: The `from` Element has been tested on the `to` Element(s).
+- tracedToDetail: the `from` Requirement has some more detailed implementation information in each `to` Requirement.
```
```suggestion
- tracedToDetail: the `from` Requirement is refined and further elaborated by each `to` Requirement, which contains more detailed implementation information.
```

stanislaw commented on model/Core/Properties/requirementRational.md:
```diff
@@ -0,0 +1,17 @@
+
+## Summary
+
+Text used to define the rational or additional information.
```
```suggestion
Text used to describe the rationale of the requirement, explaining the reasoning or justification behind its formulation.
```

stanislaw commented on model/Core/Properties/devLifeCycleStage.md:
```diff
@@ -0,0 +1,17 @@
+
+## Description
+
+There are requirements that apply to the design phase of a product, that are aimed at the development, to the build environment, give instructions for the decommissioning phase etc.
```
```suggestion
The development lifecycle stage specifies the phase or phases of a product's lifecycle to which a given requirement applies. These may include requirements that govern the design phase, guide development activities, define constraints for the build environment, provide instructions for the decommissioning phase, etc.
```

stanislaw commented on model/Core/Properties/devLifeCycleStage.md:
```diff
@@ -0,0 +1,17 @@
+SPDX-License-Identifier: Community-Spec-1.0
+
+# devLifeCycleStage
```
I am not a native English speaker but I think we can simply do `devLifecycleStage` because lifecycle is really one word.

bact replied:
I'm not a native English speaker as well and in general I don't have a preference between "LifeCycle" and "Lifecycle" -- but as we already have [/Core/LifecycleScopedRelationship](https://github.com/spdx/spdx-3-model/blob/develop/model/Core/Classes/LifecycleScopedRelationship.md), we should keep using "Lifecycle".

stanislaw reviewed (commented):
It is a great, well-focused PR. I really welcome this change!

I only left some smaller suggestions, not changing the overall structure of what is proposed.

(Sorry, I missed to join the FuSa meeting due to the timezone change 🥲 )

nicpappler committed:
Update model/Core/Classes/Requirement.md

Co-authored-by: Stanislav Pankevich <s.pankevich@gmail.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update model/Core/Classes/Requirement.md

Co-authored-by: Stanislav Pankevich <s.pankevich@gmail.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update model/Core/Properties/devLifeCycleStage.md

Co-authored-by: Stanislav Pankevich <s.pankevich@gmail.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

bact committed:
Fix typo as suggested

Co-authored-by: Stanislav Pankevich <s.pankevich@gmail.com>
Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

stevenc-stb committed:
Fix for validation

This is a fix for the .md  as per https://github.com/spdx/spdx-3-model/blob/develop/docs/format.md#properties

Signed-off-by: stevenc-stb <steven@smarttalkbeacon.com>

nicpappler committed:
Update and rename devLifeCycleStage.md to devLifecycleStage.md

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Rename devLifeCycleStage to devLifecycleStage

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Fix typo in requirement rationale documentation

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update model/Core/Vocabularies/RelationshipType.md

Co-authored-by: Stanislav Pankevich <s.pankevich@gmail.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update ContentIdentifierType.md vocabulary entries

Replaced 'reqUID' with 'requirementUUID' for clarity.

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update requirementUUID type to Software/ContentIdentifier

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Update range type for requirementUUID property

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

stanislaw committed:
Bring back the model/Core/Properties/context.md file

nicpappler committed:
Merge pull request #1 from stanislaw/stanislaw/profile_safety

Bring back the model/Core/Properties/context.md file

nicpappler committed:
Change type from Software/ContentIdentifier to ExternalIdentifier

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

nicpappler committed:
Change range type from Software/ContentIdentifier to ExternalIdentifier

Signed-off-by: Nicole Pappler <nicole@alektometis.com>

stanislaw committed:
Remove changes to ContentIdentifierType, add changes to ExternalIdentifierType

nicpappler committed:
Merge pull request #2 from stanislaw/stanislaw/content_identifiers

Remove changes to ContentIdentifierType, add changes to ExternalIdentifierType

stanislaw committed:
model/Core/Vocabularies/RelationshipType.md: remove verifiedBy, add implementedBy

This verifiedBy will come with the next PRs. Not needed here.

nicpappler committed:
Merge pull request #3 from stanislaw/stanislaw/remove_verified_by

model/Core/Vocabularies/RelationshipType.md: remove verifiedBy

stanislaw commented on model/Core/Properties/requirementRationale.md:
```diff
@@ -0,0 +1,17 @@
+
+## Description
+
+Additional detail used to define the reason/rationale, why this requirement is there or give some additional information, needed to understand the requirement. Usually less formal than the requirement wording of the RequirementStatement.
```
```suggestion
Additional detail used to define the reason or justification for the existence of the requirement. The rationale is usually less formal than the wording of the requirement statement itself.
```

stanislaw replied:
This my last review comment became Outdated due to the other changes in this PR. I think it is a good improvement, so I restored it here.

nicpappler committed:
Update model/Core/Properties/requirementRationale.md

Co-authored-by: Stanislav Pankevich <s.pankevich@gmail.com>
Signed-off-by: Nicole Pappler <nicole@alektometis.com>

kestewart reviewed (approved):
Matches my understanding of what we discussed. 

Merging this into the spdx:profile-safety branch.
